tools/docker/frontend.py

```diff
@@ -1,18 +1,93 @@
 import argparse
-import subprocess
+import logging
 import os
+import re
+import subprocess
+import sys
+
+from six import iteritems
 
 here = os.path.abspath(os.path.dirname(__file__))
 wpt_root = os.path.abspath(os.path.join(here, os.pardir, os.pardir))
 
-def build(*args, **kwargs):
+logger = logging.getLogger()
+
+
+def build(tag="wpt:local", *args, **kwargs):
     subprocess.check_call(["docker",
                            "build",
                            "--pull",
-                           "--tag", "wpt:local",
+                           "--tag", tag,
                            here])
 
 
+def parser_push():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--tag", action="store",
+                        help="Tag to use (default is taken from .taskcluster.yml)")
```

Comment:
I'm a little hesitant about this. Tags *can* be overwritten on Docker Hub, so if someone (with the right permissions) runs `wpt docker-push` without bumping the version in the tag first, we'd end up changing the currently used Docker image on Taskcluster **without** having to land any change in code.

I'd much prefer defaulting to current version + 0.01, and either update the yaml files automatically or prompt the user to to do that (if editing is hard with pyyaml).

Reply:
Oh wow, I just assumed… and apparently you need to buy enterprise to fix this. Anyway I added a check on whether the tag already exists or not. I rather like the workflow where it reads the tag from the yaml files, and as you note rewriting the files with the updated data is tricky to do in a nice way.